From dc3466b2465e16ddadfa75dbc06b77f115f6831d Mon Sep 17 00:00:00 2001
From: Abu Kashem <akashem@redhat.com>
Date: Fri, 29 Apr 2022 17:51:54 -0400
Subject: [PATCH] UPSTREAM: <carry>: fix [sig-auth] ServiceAccounts no
 secret-based service account token should be auto-generated

OpenShift-Rebase-Source: a03143879c0

UPSTREAM: <carry>: fix [sig-auth] ServiceAccounts no secret-based service account token should be auto-generated
---
 test/e2e/auth/service_accounts.go | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/test/e2e/auth/service_accounts.go b/test/e2e/auth/service_accounts.go
index 654600d7e32..0173d999f4f 100644
--- a/test/e2e/auth/service_accounts.go
+++ b/test/e2e/auth/service_accounts.go
@@ -64,6 +64,17 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 			time.Sleep(10 * time.Second)
 			sa, err := f.ClientSet.CoreV1().ServiceAccounts(f.Namespace.Name).Get(ctx, "default", metav1.GetOptions{})
 			framework.ExpectNoError(err)
+
+			// TODO: Ignore the image pull secret that OpenShift sometimes creates and adds to the list of Secrets.
+			// TODO: This patch can be removed once OpenShift stops adding the pull secret to the list of secrets in 4.16.
+			secrets := sa.DeepCopy().Secrets
+			sa.Secrets = nil
+			for _, s := range secrets {
+				if strings.HasPrefix(s.Name, "default-dockercfg") {
+					continue
+				}
+				sa.Secrets = append(sa.Secrets, s)
+			}
 			gomega.Expect(sa.Secrets).To(gomega.BeEmpty())
 		}
 	})
-- 
2.46.0

